src/Shared/FileUtilitiesRegex.cs

```diff
@@ -34,5 +34,180 @@ internal static class FileUtilitiesRegex
             new Regex(
                 string.Format(CultureInfo.InvariantCulture, @"{0}$", s_baseUncPattern),
                 RegexOptions.Compiled);
+
+
+        /// <summary>
+        /// Indicates whether the specified string follows the pattern drive pattern: "<drive letter>:"
+        /// </summary>
+        /// <param name="pattern"></param>
+        /// <returns></returns>
+        internal static bool IsDrivePattern(string pattern)
+        {
+            // Format must be two characters long: "<drive letter>:"
+            return pattern.Length == 2 &&
+                DoesStartWithDrivePattern(pattern);
+        }
+
+        /// <summary>
+        /// Indicates whether the specified string starts with the drive pattern: "<drive letter>:".
+        /// </summary>
+        /// <param name="pattern"></param>
+        /// <returns></returns>
+        internal static bool DoesStartWithDrivePattern(string pattern)
+        {
+            // Format dictates a length of at least 2,
+            // first character must be a letter,
+            // second character must be a ":"
+            return pattern.Length >= 2 &&
+                ((pattern[0] >= 'A' && pattern[0] <= 'Z') || (pattern[0] >= 'a' && pattern[0] <= 'z')) &&
+                pattern[1] == ':';
+        }
+
+        /// <summary>
+        /// Indicates whether the specified file-spec comprises exactly "\\<server>\<path>" (with no trailing characters).
+        /// </summary>
+        /// <param name="pattern"></param>
+        /// <returns></returns>
+        internal static bool IsUncPattern(string pattern)
+        {
+            if(!MeetsUncPatternMinimumRequirements(pattern))
+            {
+                return false;
+            }
+
+            bool prevCharWasSlash = true;
+            bool hasSubfolder = false;
+            for (int i = 2; i < pattern.Length; i++)
+            {
+                if (pattern[i] == Path.DirectorySeparatorChar ||
```

Comment:
Probably worth a comment that real UNC paths should only be backslashes but the previous pattern accepted both so this function does too.

Reply:
Comment added